java/src/main/java/org/rocksdb/RocksDB.java

```diff
@@ -1435,6 +1435,25 @@ public void deleteRange(final ColumnFamilyHandle columnFamilyHandle, final Write
         endKey.length, columnFamilyHandle.nativeHandle_);
   }
 
+  /**
+   * Delete files which are entirely in the given range for default column family.
+   * Could leave some keys in the range which are in files which are not
+   * entirely in the range. Also leaves L0 files regardless of whether they're
+   * in the range.
+   * Snapshots before the delete might not see the data in the given range.
+   *
+   * @param beginKey
+   *          First key to delete within database (included)
+   * @param endKey
+   *          Last key to delete within database (included)
+   *
+   * @throws RocksDBException
+   *           thrown if error happens in underlying native library.
+   */
+  public void deleteFilesInRange(final byte[] beginKey, final byte[] endKey) throws RocksDBException {
```

Comment:
Why are the offsets and lengths not also exposed publicly?

Reply:
👍 will add more interface to expose them